### ISSUE TITLE
Anonymous controller doesn't include itself in routes

### ISSUE THREAD

**Issue by BMorearty** [rspec/rspec-rails#998]
TL;DR Anonymous controllers can't redirect to themselves, which causes problems with some specs. I have written a potential fix but it has a side-effect that may or may not be good. 

Longer version:

I'm trying to write a test relating to SSL, so I've defined an anonymous controller along these lines:

``` ruby
controller do
  force_ssl only: :index
end

it "redirects to ssl" do
  get :index
  expect(response.location).to start_with("https://")
end
```

But the `get` fails because the anonymous controller can't redirect to itself. In `controller_example_group.rb`, rspec-rails resets the example group's route set to a brand new one containing only one route: `resources :anonymous`. But inside the controller, it's still using the application's route set.

On my fork I wrote [some code that might fix this problem](https://github.com/BMorearty/rspec-rails/commit/9cf4714d8d9ae0090a9021e663e93e3d00e58535), but it introduces another problem (cofirmed by running the rspec-rails specs). My code redefines the controller's `_routes` method to return the same routes that are in use by the example group. This seems like the right thing to do. They are in sync with each other, and are in fact the same instance of `RouteSet`.

However, if the anonymous controller is a subclass of a controller with a `before_filter`--and that filter redirects to something in the application route set--it explodes. Because the app route set is no longer there in the controller.

Here's the failing spec:

```

/Users/Brian/.rvm/rubies/ruby-2.1.1/bin/ruby -S bundle exec cucumber
Using the default profile...
....................F.........................................................................................................................................................................................................................

(::) failed steps (::)

expected "F\n\nFailures:\n\n  1) ApplicationController redirects to the login page\n     Failure/Error: redirect_to login_url\n     ActionController::RoutingError:\n       No route matches {:controller=>\"sessions\", :action=>\"new\"}\n

...

+Failures:
+
+  1) ApplicationController redirects to the login page
+     Failure/Error: redirect_to login_url
+     ActionController::RoutingError:
+       No route matches {:controller=>"sessions", :action=>"new"}
+     # /Users/Brian/.rvm/gems/ruby-2.1.1/gems/actionpack-3.1.3/lib/action_dispatch/routing/route_set.rb:465:in `raise_routing_error'
+     # /Users/Brian/.rvm/gems/ruby-2.1.1/gems/actionpack-3.1.3/lib/action_dispatch/routing/route_set.rb:461:in `rescue in generate'
+     # /Users/Brian/.rvm/gems/ruby-2.1.1/gems/actionpack-3.1.3/lib/action_dispatch/routing/route_set.rb:453:in `generate'
+     # /Users/Brian/.rvm/gems/ruby-2.1.1/gems/actionpack-3.1.3/lib/action_dispatch/routing/route_set.rb:494:in `generate'
+     # /Users/Brian/.rvm/gems/ruby-2.1.1/gems/actionpack-3.1.3/lib/action_dispatch/routing/route_set.rb:519:in `url_for'
+     # /Users/Brian/.rvm/gems/ruby-2.1.1/gems/actionpack-3.1.3/lib/action_dispatch/routing/url_for.rb:147:in `url_for'
+     # /Users/Brian/.rvm/gems/ruby-2.1.1/gems/actionpack-3.1.3/lib/action_dispatch/routing/route_set.rb:201:in `login_url'
+     # ./spec/controllers/application_controller_spec.rb:10:in `index'
+     # /Users/Brian/.rvm/gems/ruby-2.1.1/gems/actionpack-3.1.3/lib/action_controller/metal/implicit_render.rb:4:in `send_action'
+
```
